src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -42,41 +46,76 @@ public static class CreateMutationBuilder
                 return inputs[inputName];
             }
 
-            IEnumerable<InputValueDefinitionNode> inputFields =
-                objectTypeDefinitionNode.Fields
-                .Where(f => FieldAllowedOnCreateInput(f, databaseType, definitions))
+            // The input fields for a create object will be a combination of:
+            // 1. Simple input fields corresponding to columns which belong to the table.
+            // 2. Complex input fields corresponding to tables having a foreign key relationship with this table.
+            List<InputValueDefinitionNode> inputFields = new();
+
+            // 1. Simple input fields.
+            IEnumerable<InputValueDefinitionNode> simpleInputFields = objectTypeDefinitionNode.Fields
+                .Where(f => IsBuiltInType(f.Type))
+                .Where(f => IsBuiltInTypeFieldAllowedForCreateInput(f, databaseType))
                 .Select(f =>
                 {
-                    if (!IsBuiltInType(f.Type))
-                    {
-                        string typeName = RelationshipDirectiveType.Target(f);
-                        HotChocolate.Language.IHasName? def = definitions.FirstOrDefault(d => d.Name.Value == typeName);
-
-                        if (def is null)
-                        {
-                            throw new DataApiBuilderException($"The type {typeName} is not a known GraphQL type, and cannot be used in this schema.", HttpStatusCode.InternalServerError, DataApiBuilderException.SubStatusCodes.GraphQLMapping);
-                        }
-
-                        if (def is ObjectTypeDefinitionNode otdn)
-                        {
-                            //Get entity definition for this ObjectTypeDefinitionNode
-                            return GetComplexInputType(inputs, definitions, f, typeName, otdn, databaseType, entities);
```

Comment:
We used to call `GetComplexInputType` for other databases as well, previously - that doesnt seem to happen in the new changes when we add the ComplextInputType for relationships. Previously, looks like the complex input type was being added for fields which had relationship directive on them.